projects/systelab-translate/src/lib/date-util/date-util.ts

```diff
@@ -87,91 +87,20 @@ export class DateUtil {
 	}
 
 	public getDateFormat(isFullYear = false): string {
-		let stringDateFormat = '';
-		switch (this.locale) {
-			case 'en-US':
-				stringDateFormat = 'M/d/yy';
-				break;
-			case 'ko-KO':
-				stringDateFormat = 'yy. M. d';
-				break;
-			case 'pl-PL':
-			case 'lt-LT':
-				stringDateFormat = 'yy-MM-dd';
-				break;
-			case 'pt-PT':
-			case 'pt-BR':
-			case 'nl-NL':
-				stringDateFormat = 'dd-MM-yy';
-				break;
-			case 'sk-SK':
-			case 'ru-RU':
-				stringDateFormat = 'd.M.yy';
-				break;
-			case 'zh-ZH':
-				stringDateFormat = 'yy-M-d';
-				break;
-			case 'de-DE':
-				stringDateFormat = 'dd.MM.yy';
-				break;
-			case 'th-TH':
-				stringDateFormat = 'd/M/yy';
-				break;
-			case 'ja-JA':
-				stringDateFormat = 'yy/MM/dd';
-				break;
-			default:
-				stringDateFormat = 'dd/MM/yy';
-				break;
-		}
+		let stringDateFormat = this.getLocalizedStringDateFormat(this.locale);
+
 		if (isFullYear) {
 			stringDateFormat = stringDateFormat.replace('yy', 'yyyy');
 			if (this.locale === 'en-US') {
-				stringDateFormat = stringDateFormat.replace('M', 'MM');
-				stringDateFormat = stringDateFormat.replace('D', 'dd');
+				stringDateFormat = stringDateFormat.replace('M', 'MM').replace('d', 'dd');
```

Comment:
If the format contains two "dd" or two "MM" the replacement will add a new d or M.
'dd/MM/yy'.replace('M', 'MM').replace('d', 'dd')  == 'ddd/MMM/yy'

Reply:
But this is only for en-US and the date format is 'M/d/yy'

Reply:
Anyway, I moved all the logic to the function returning the format string to avoid this issues